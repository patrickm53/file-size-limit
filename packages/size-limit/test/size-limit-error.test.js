let SizeLimitError = require('../size-limit-error')
let createReporter = require('../create-reporter')

function print (err) {
  let stderr = ''
  let process = {
    stderr: {
      write  (str) {
        stderr += str
      }
    }
  }
  let processor = createReporter(process, false)
  processor.error(err)
  return stderr
}

it('has mark', () => {
  let err = new SizeLimitError('noPackage')
  expect(err.name).toEqual('SizeLimitError')
})

it('has start', () => {
  let err = new SizeLimitError('noPackage')
  expect(err.stack).toContain('size-limit-error.test.js')
})

it('has message', () => {
  let err = new SizeLimitError('noPackage')
  expect(err.message).toContain('Create npm package')
})

it('has error for unknown option', () => {
  let err = new SizeLimitError('missedPlugin', 'webpack', 'file')
  expect(print(err)).toMatchSnapshot()
})

it('has error for unknown entry', () => {
  let err = new SizeLimitError('unknownEntry', 'admin')
  expect(print(err)).toMatchSnapshot()
})
