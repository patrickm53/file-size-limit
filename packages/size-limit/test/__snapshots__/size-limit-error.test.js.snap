// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has error for unknown entry 1`] = `
"[41m[30m ERROR [39m[49m [31mSize Limit didnâ€™t find [33madmin[39m[31m entry in custom Webpack config[39m
"
`;

exports[`has error for unknown option 1`] = `
"[41m[30m ERROR [39m[49m [31mAdd [33m@size-limit/webpack[39m[31m plugin to Size Limit[39m
"
`;
